dataplicity/portforward.py

```diff
@@ -317,9 +326,19 @@ def open(self, m2m_port, service=None, port=None):
             return
         service.connect(m2m_port)
 
-    def redirect_service(self, m2m_port, device_port):
-        service = Service(
-            manager=self, name='port-{}'.format(device_port),
-            port=device_port, host='127.0.0.1'
-        )
-        service.connect(m2m_port)
+    def redirect_port(self, m2m_port, device_port):
+        # we need to store the reference to the Service somewhere so that
+        # when the Connection starts in thread it wouldn't loose the value
+        # of service variable. However, we have to remember that there may
+        # be numerous connections to the same local port.
+        # for instance, one could be ssh'ed into a machine twice, so we
+        # shan't confuse these two connections.
+        # therefore, an easy way is to store these in a dict, so that the
+        # lookup would be quick
+        #
+        Connection(
+            close_event=self.close_event,
+            connection_id=-1,
```

Comment:
The `connection_id` is used as a key to `Service._connections`. Having it as a constant means that a second connection will overwrite the first.

It might be that that `Service._connections` isn't actually required -- given that the threads are started and maintain their own reference.

If that's the case, feel free to remove the relevant code. Suspect it will simplify a fair bit.

Reply:
hmm I did have two simultaneous connections because I was doing htop and scp at the same time. but I'll inspect the code after your hint now. Thanks!

Reply:
ok after the discussion I checked the code with connection_id removed and everything works!